server/controllers/finance/fiscal.js

```diff
@@ -590,23 +594,25 @@ function closing(req, res, next) {
         const credit = value >= 0 ? Math.abs(value) : 0;
 
         const resultParams = [
-          projectId,                      // project_id
-          fiscal.id,                      // fiscal_year_id
-          fiscal.number_of_months + 1,    // period_id
-          fiscal.end_date,                // date : the last date of the fiscal year
-          'Ecriture du result lors de la cloture',
+          db.bid(uuid.v4()),
           accountId,
-          debit, credit,                  // debit and credit
-          debit, credit,                  // debit_equiv and credit_equiv in enterprise currency
-          currencyId,                     // enterprise currency because data came from period total
-          userId,                         // user id
+          debit,
+          credit,
+          voucher.uuid,
+          voucherDocumentUuid,
         ];
 
-        transaction.addQuery(sqlInsertJournal, resultParams);
+        transaction.addQuery(sqlInsertVoucherItem, resultParams);
       }
 
-      // lock the fiscal year
-      transaction.addQuery('UPDATE fiscal_year SET locked = 1 WHERE id = ?;', [id]);
+      // update the period to the period N+1 (13)
+      const sqlUpdateJournalPeriod = 'UPDATE posting_journal SET period_id = ? WHERE reference_uuid = ?';
+
+      // post voucher and close the fiscal year
+      transaction
```

Comment:
When a fiscal year is closed, exploitation accounts are `solded`. This operation of ,solding exploitation account and put the result in a corespondent account, is an official accounting operation, so it must be in the journal.

There is no issue with that logic, but the idea is to write into the Journal via a voucher, and these data must be posted normally into the general ledger if all things are validate by the organization.